src/System.Net.Security/src/System/Net/SslSessionsCache.cs

```diff
@@ -12,92 +13,80 @@ namespace System.Net.Security
     internal static class SslSessionsCache
     {
         private const int CheckExpiredModulo = 32;
-        private static Hashtable s_CachedCreds = new Hashtable(32);
+        private static readonly ConcurrentDictionary<SslCredKey, SafeCredentialReference> s_cachedCreds =
+            new ConcurrentDictionary<SslCredKey, SafeCredentialReference>();
 
         //
         // Uses certificate thumb-print comparison.
         //
-        private struct SslCredKey
+        private struct SslCredKey : IEquatable<SslCredKey>
         {
-            private byte[] _CertThumbPrint;
-            private int _AllowedProtocols;
-            private bool _isServerMode;
-            private EncryptionPolicy _EncryptionPolicy;
-            private readonly int _HashCode;
+            private readonly byte[] _thumbPrint;
+            private readonly int _allowedProtocols;
+            private readonly EncryptionPolicy _encryptionPolicy;
+            private readonly bool _isServerMode;
```

Comment:
The last three fields here (`_allowedProtocols`, `_encryptionPolicy`, and `_isServerMode`) could be collapsed into a single `int` field for a smaller struct / improved enumeration perf, but I didn't have a chance to get to it.